docs/advanced_topics/documents/title_generation_on_upload.md

```diff
@@ -32,82 +32,82 @@ See MDN for more information about [custom JavaScript events](https://developer.
 
 ## Code examples
 
+For each example below, create the specified external JavaScript file in your app’s static directory, such as static/js/, and reference it in the wagtail_hooks.py file.
+
 ### Adding the file extension to the start of the title
 
 ```python
 # wagtail_hooks.py
 from django.utils.safestring import mark_safe
-
 from wagtail import hooks
 
-
 @hooks.register("insert_global_admin_js")
 def get_global_admin_js():
-    return mark_safe(
-    """
-    <script>
-    window.addEventListener('DOMContentLoaded', function () {
-        document.addEventListener('wagtail:documents-upload', function(event) {
-            var extension = (event.detail.filename.match(/\.([^.]*?)(?=\?|#|$)/) || [''])[1];
-            var newTitle = '(' + extension.toUpperCase() + ') ' + (event.detail.data.title || '');
-            event.detail.data.title = newTitle;
-        });
+    return mark_safe('<script src="{% static \'js/title_with_extension.js\' %}"></script>')
```

Comment:
I don't think putting a Django template tag inside a string like this is going to work - it'll just output the tag verbatim, rather than evaluating it. Please test this code.

There's an example of outputting a `<script src="...">` tag from a hook here: https://docs.wagtail.org/en/stable/reference/hooks.html#insert-editor-js

Reply:
Will do, thanks!